ecosystem/sep-0024.md

```diff
@@ -190,6 +192,127 @@ The deposit flow can only be fulfilled if the Stellar `Account` has established
 5. `Anchor` detects the trust line creation in the `Account`. If the asset is `AUTH_REQUIRED`, `Anchor` approves the new trust line.
 6. `Anchor` proceeds with the deposit flow.
 
+##### Claimable Balances
+
+[Claimable Balances](https://www.stellar.org/blog/protocol-14-improvements) enables payments to accounts that do not have a trust line for the asset being sent.
+This feature provides less friction for Stellar accounts who wish to make an offchain deposit to anchors without making a trust line before the deposit is complete.
+Anchors will no longer have to wait during the time when the account doesn't have a trust line. Instead, anchors can make the payment using ClaimableBalance operation and the stellar account (wallet user) can collect the funds at their own convenience.
+
+**NOTE**: The Stellar Account will still need to establish trust line to the asset before claiming the outstanding balance.
+
+**Wallets**: To support claimable balances wallets must
+
+* Send `claimable_balance_supported=true` in their POST /transactions/deposit/interactive request.
+* Periodically poll for account's available claimable balances.
+* Provide a UI that allows users to claim a claimable balance. (The wallet UI should allow users to trust the asset of the claimable balance in order to claim)
+
+**Anchors**: To support claimable balances anchors must
+
+* Accept the `claimable_balance_supported` field in their `POST /transactions/deposit/interactive` request and submit the deposit transaction as a Claimable Balance operation to Stellar account that don't yet have a trust-line to the asset they have requested.
+* Add `claimable_balance_id` to their [deposit transactions](####Fields-for-deposit-transactions)
+
+**Anchors and Wallets**: Both anchors and wallets must also support the aforementioned ["stellar account doen't have a trust line for that asset."](#stellar-account-doesnt-trust-asset).
+Wallets need to be interoperable with anchors that have not adopted the [claimable balance feature](#Claimable-Balances) and vice versa.
+
+##### Wallet Claimable Balance Flow
+
+Wallets will still follow the [typical deposit flow](#Basic-Wallet-Implementation).
+Wallets will provide `claimable_balance_supported=true` field and value in their POST parameters for `/transactions/deposit/interactive`.
+
+If the anchor we're depositing to **supports** claimable balances.
```

Comment:
Wallets unfortunately won't know if claimable balances are supported unless they hard-code it per-anchor. So i think we need to rewrite this **supports**/**doesn't support** section from the perspective that the wallet doesn't know.

That would involve sending `claimable_balance_supported: true`, then polling the service for `pending_trust`. If the status goes to `completed` without ever moving to `pending_trust`, the wallet knows to check for claimable balances, and the `claimable_balance_id` should be populated.

Reply:
The way I intended to frame `If the anchor we're depositing to **supports** claimable balances.` and `If the anchor we're depositing to ** doesn't support** claimable balances.` is from the wallet's perspective they do not know if the anchor their issuing a deposit to is `claimable_balance_supported` or not, they're just shooting their shot. 
Let me explain using the demo wallet implementation as an example as far as I understand it.

1. Wallet issues a deposit for a user (no trustline to depositing asset) and sends the `claimable_balance_supported` POST request param.
2. The wallet listens for the anchor's callback for the transaction status.
3. Until the transaction is completed we communicate its pending.
From the Wallet logs
```
To collect the interactive information we launch the interactive URL in a frame or webview, and await payment details from a postMessage callback
Transaction status pending...
Transaction status complete
```
If the anchor supports claimable balances it will do all the things it needs to do to process the claimable balance transaction.
Then mark it as complete

If the anchor doesnt support claimable balances (lets say it ignores the param) 
The wallet will keep listening to the transaction status from the callback.
```
To collect the interactive information we launch the interactive URL in a frame or webview, and await payment details from a postMessage callback
Transaction status pending...
Transaction status pending...
Transaction status pending...
```
On the interactive window from `sep24/transaction/more_info` we'll see the status `waiting for trust-line to be established`
from there the wallet is busy waiting (like before) but this time its because it `pending_trust`  until the user makes that trustline.
I reproduced this by sending `claimable_balance_supported: false` from my wallet and letting polaris(`316-claimable-balance-attempt2`) do it's trustline flow.

This is the wallet's callback loop 
```
window.onmessage = ({ data: { transaction } }) => {
      if (transaction.status === 'completed') {
        this.updateAccount()
        this.logger.instruction('Transaction status complete')
        finish()
      } else {
        this.logger.instruction('Transaction status pending...')
        setTimeout(() => {
          const urlBuilder = new URL(transaction.more_info_url)
          urlBuilder.searchParams.set('callback', 'postMessage')

          popup.location.href = urlBuilder.toString()
        }, 1000)
      }
    }
```